crypto/s2n_certificate.c

```diff
@@ -883,3 +885,34 @@ int s2n_cert_get_x509_extension_value(struct s2n_cert *cert, const uint8_t *oid,
 
     return S2N_SUCCESS;
 }
+
+S2N_RESULT s2n_cert_get_cert_description(X509 *cert, struct s2n_cert_description *description)
+{
+    X509_NAME *issuer_name = X509_get_issuer_name(cert);
+    RESULT_ENSURE_REF(issuer_name);
+
+    X509_NAME *subject_name = X509_get_subject_name(cert);
+    RESULT_ENSURE_REF(subject_name);
+
+    if (X509_NAME_cmp(issuer_name, subject_name) == 0) {
+        description->self_signed = true;
+    } else {
+        description->self_signed = false;
+    }
+
+#if defined(LIBRESSL_VERSION_NUMBER) && (LIBRESSL_VERSION_NUMBER < 0x02070000f)
+    RESULT_ENSURE_REF(cert->sig_alg);
+    description->signature_nid = OBJ_obj2nid(cert->sig_alg->algorithm);
+#else
+    description->signature_nid = X509_get_signature_nid(cert);
+#endif
+
+    DEFER_CLEANUP(EVP_PKEY *pubkey = X509_get_pubkey(cert), EVP_PKEY_free_pointer);
+    RESULT_ENSURE_REF(pubkey);
+
+    RESULT_GUARD_OSSL(OBJ_find_sigid_algs(description->signature_nid,
```

Comment:
Can you tell me more about OBJ_find_sigid_algs? I get that you are populating the signature_digest_nid, but I'm not sure why that's coming from the signature_nid rather than the cert. This line might need to be commented.